### PR TITLE
Add bindings to WebWorker

### PR DESCRIPTION
**Pull request by OCamlPro-Henry**
This code was developped for the OCaml MOOC. An embedding of the
toplevel into a WebWorker should follow in another PR.

### CODE REVIEW

File: lib/worker.ml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    invalid_arg "Worker.import_scripts is undefined";
+  Unsafe.fun_call
+    (Unsafe.global##importScripts)
+    (Array.map (fun s -> Unsafe.inject @@ string s) (Array.of_list scripts))
+
+let set_onmessage handler =
+  if Unsafe.global##onmessage == undefined then
+    invalid_arg "Worker.onmessage is undefined";
+  let js_handler (ev : 'a messageEvent Js.t) =
+    Lwt.async (fun () -> handler (ev##data)) in
```
Comment (hhugo) [inline lib/worker.ml]: I would remove `Lwt.async` as it doesn't add any feature.
I hope to add support for core & async soon.

Comment (OCamlPro-Henry) [inline lib/worker.ml]: Indeed. Fixed.

**Comment by hhugo** [comment]
Would you mind renaming `worker` to `webWorker`  (similarly to `mutationObserver`, `xmlHttpRequest`, `webSockets`, ...)

File: lib/worker.mli [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ *)
+
+(** Low-level bindgins to javascript Web Workers.
+
+    See [https://developer.mozilla.org/en-US/docs/Web/API/Worker]
+    and [https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers].
```
Comment (Drup) [inline lib/worker.mli]: The ocamldoc formatting for links is `{{:link}text}` (iirc).

**Comment by OCamlPro-Henry** [comment]
I don't mind renaming but the javascript class is named `Worker`.

**Comment by hhugo** [comment]
Indeed. Ignore my request then

File: lib/worker.mli (hunk 2 above)
Comment (OCamlPro-Henry) [inline lib/worker.mli]: Fixed. Thanks fot spotting this.

**Comment by hhugo** [comment]
I'm now tempted to include this in the next release..
Any thought ?

**Comment by hhugo** [comment]
Thanks

**Pull request merged**